plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy

```diff
@@ -304,15 +302,43 @@ class AzBatchService implements Closeable {
         return servicePrincipalBasedCred
     }
 
+    protected BatchCredentials createBatchCredentialsWithManagedIdentity() {
+        log.debug "[AZURE BATCH] Creating Azure Batch client using Managed Identity credentials"
+
+        final batchEndpoint = "https://batch.core.windows.net/"
+        final authenticationEndpoint = "https://login.microsoftonline.com/"
+
+        final clientId = config.managedIdentity().clientId
+        final credentialBuilder = new DefaultAzureCredentialBuilder()
+        if( clientId )
+            credentialBuilder.managedIdentityClientId(clientId)
+        final credential = credentialBuilder.build()
+        final token = credential.getTokenSync( new TokenRequestContext() )
+
+        return new BatchApplicationTokenCredentials(
+                config.batch().endpoint,
+                clientId,
+                token.getToken(),
+                null,
+                batchEndpoint,
+                authenticationEndpoint
+        )
```

Comment:
@vsmalladi I adapted this from your C# solution [here](https://stackoverflow.com/questions/76326790/how-to-use-user-managed-identity-to-access-azure-batchclient-programatically) but I'm pretty sure I'm missing something here. In the C# SDK there is `BatchTokenCredentials` which only requires the account URL and token, but in the Java SDK there is [BatchApplicationTokenCredentials](https://learn.microsoft.com/en-us/java/api/com.microsoft.azure.batch.auth.batchapplicationtokencredentials?view=azure-java-stable) and BatchUserTokenCredentials which both require more arguments.

Perhaps you can enlighten us on how to set up these batch credentials with the managed identity